File: engine/model/src/main/scala/net/server/channel/packet/interaction/BuyPlayerInteractionPacket.scala
package net.server.channel.packet.interaction

class BuyPlayerInteractionPacket(private var _mode: Byte, private var _itemId: Int, private var _quantity: Short) extends BasePlayerInteractionPacket(_mode = _mode) {
  def itemId: Int = _itemId

  def quantity: Short = _quantity
}

File: engine/engine-model/src/main/java/net/server/channel/packet/interaction/BuyPlayerInteractionPacket.java
package net.server.channel.packet.interaction;

public class BuyPlayerInteractionPacket extends BasePlayerInteractionPacket {
   private final Integer itemId;

   private final Short quantity;

   public BuyPlayerInteractionPacket(Byte mode, Integer itemId, Short quantity) {
      super(mode);
      this.itemId = itemId;
      this.quantity = quantity;
   }

   public Integer itemId() {
      return itemId;
   }

   public Short quantity() {
      return quantity;
   }
}
